src/joystick/iphoneos/SDL_mfijoystick.m

```diff
@@ -1535,9 +1556,12 @@ -(void)cleanup
 SDL_bool IOS_SupportedHIDDevice(IOHIDDeviceRef device)
 {
     if (is_macos11()) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunguarded-availability-new"
```

Comment:
Ugh, these look ugly in general..

For this particular one though: Does the following one help, instead?
(assuming the patch is correct - not tested...)
```diff
diff --git a/src/joystick/iphoneos/SDL_mfijoystick.m b/src/joystick/iphoneos/SDL_mfijoystick.m
index 605651f..514e482 100644
--- a/src/joystick/iphoneos/SDL_mfijoystick.m
+++ b/src/joystick/iphoneos/SDL_mfijoystick.m
@@ -542,2 +542,5 @@ SDL_AppleTVRemoteRotationHintChanged(void *udata, const char *name, const char *
 #if defined(__MACOSX__)
+#if defined(MAC_OS_X_VERSION_10_13)
+#define is_macos11() @available(macOS 11.0,*)
+#else
 static int is_macos11(void)
@@ -547,2 +550,3 @@ static int is_macos11(void)
 #endif
+#endif
 
```

Reply:
Unfortunately `@available` doesn't seem to work inside a macro:
```
/Users/runner/work/SDL/SDL/src/joystick/iphoneos/SDL_mfijoystick.m:566:10: warning: @available does not guard availability here; use if (@available) instead [-Wunsupported-availability-guard]
    if (!is_macos11()) {
         ^
/Users/runner/work/SDL/SDL/src/joystick/iphoneos/SDL_mfijoystick.m:553:22: note: expanded from macro 'is_macos11'
#define is_macos11() @available(macOS 11.0,*)
                     ^
1 warning generated.
```

Reply:
Ew..

The thing with the diagnostic pragma is that older clang might not
have it and the pragma itself might emit a warning instead. Don't know
what's a good solution here..